tools/clang/lib/Sema/SemaOverload.cpp

```diff
@@ -10914,7 +10914,18 @@ bool Sema::buildOverloadedCallSet(Scope *S, Expr *Fn,
 #ifndef NDEBUG
   if (ULE->requiresADL()) {
     // To do ADL, we must have found an unqualified name.
-    assert(!ULE->getQualifier() && "qualified name with ADL");
+    // HLSL Change Begins
+    //
+    // We do want to allow argument-dependent lookup for intrinsic
+    // function names inside the "vk" namespace (which are by definition
+    // qualified names).
+    bool isVkNamespace =
+        ULE->getQualifier() &&
+        ULE->getQualifier()->getKind() == NestedNameSpecifier::Namespace &&
+        ULE->getQualifier()->getAsNamespace()->getName() == "vk";
+
+    assert((!ULE->getQualifier() || isVkNamespace) && "qualified name with ADL");
```

Comment:
you might choose to update the message. "non-vk qualified name with ADL"?

Reply:
Done